### PR TITLE
Raise error and fail fast if missing pk in record instead of just logging

### PR DESCRIPTION
**Pull request by asaf-erlich**
# Description of change
The tap will fail in the target if the record is missing the key, and this would make it fail much faster. See support ticket https://stitchdata.atlassian.net/browse/SUP-1862

Only worry I have is this was logging the full record, and the exception includes the full record, should I remove that?

# Manual QA steps
 - Ran the cloned connection to reproduce the error, resulting in failing on the first record instead of doing lots of syncing only to fail much later when it gets to the target
 
# Risks
 - 
 
# Rollback steps
 - revert this branch

### CODE REVIEW

File: tap_google_search_console/sync.py [hunk 1]
```diff
@@ -217,27 +214,23 @@ def sync_endpoint(client, #pylint: disable=too-many-branches
...
             LOGGER.info('Number of raw data records: 0')
-        # LOGGER.info('transformed_data = {}'.format(transformed_data))  # TESTING, comment out
         if not transformed_data or transformed_data is None:
             LOGGER.info('xxx NO TRANSFORMED DATA xxx')
             return 0 # No data results
         for record in transformed_data:
             for key in id_fields:
                 if not record.get(key):
-                    LOGGER.info('Missing key {} in record: {}'.format(key, record))
+                    raise ValueError('Missing key {} in record: {}'.format(key, record))
```
Comment (dmosorast) [inline tap_google_search_console/sync.py:233]: Yes, absolutely remove the full record. If there's a non-PII value that can help with context, that can likely be logged, but full record is almost certainly not ok due to GDPR/PII, etc concerns.

**Review by dmosorast — CHANGES_REQUESTED** [review]

File: tap_google_search_console/sync.py (hunk 1 above)
Comment (asaf-erlich) [reply tap_google_search_console/sync.py:233]: Is it alright if I print all the pk fields?

Comment (asaf-erlich) [reply tap_google_search_console/sync.py]: Updated to only print the primary keys with values.

Comment (asaf-erlich) [reply tap_google_search_console/sync.py]: And I tested it after reproducing the error by selecting false in menagerie for this field again.

**Review by dmosorast — APPROVED** [review]

**Pull request merged**